### ISSUE TITLE
Inaccurate error message for fields with provided inappropriate value

### ISSUE THREAD

**Issue by max-moser** [inveniosoftware/invenio-app-rdm#898]
Package version (if known): 5.0.0.dev0

## Describe the bug
When uploading metadata that *does have* a value for a required field but with some kind of invalid value, e.g. an integer or `null` instead of a valid date for `publication_date`, the provided error message on publication is inaccurate.
Note that this is also true for fields where the supplied value isn't in the configured controlled vocabulary, e.g. `related_identifiers.relation_type: "asdf"`

## Steps to Reproduce
1. Upload some invalid metadata via the REST API
2. Try to publish the newly created draft
3. See error

## Expected behavior

The error message should state that the provided value is invalid, rather than stating that there is no value provided at all.

## Additional context

Error message:
`{"status": 400, "message": "A validation error occurred.", "errors": [{"field": "metadata.publication_date", "messages": ["Missing data for required field."]}]}`

Metadata:
```
{
  "files": {
    "enabled": false
  },
  "metadata": {
    "resource_type": {
      "id": "other"
    },
    "title": "Test Record",
    "description": "A test record with invalid publication date",
    "creators": [
      {
        "person_or_org": {
          "given_name": "Max",
          "family_name": "Moser",
          "type": "personal"
        },
        "affiliations": [
          {
            "name": "TU Wien",
            "identifiers": [
              {
                "scheme": "ror",
                "identifier": "04d836q62"
              }
            ]
          }
        ]
      }
    ],
    "publisher": "TU Data",
    "publication_date": 2020
  }
}
```

**Comment by Glignos** [comment]
This seems like expected behavior since, in the first reply, when the draft is created, the error is correctly returned ```"errors": [{"field": "metadata.publication_date", "messages": ["Not a valid string."]}]}%```.
We are not storing the erroneous value, since the schema would also not allow it, so then when you attempt to publish the draft it returns a validation error since we have only partially saved it in the first request.
@lnielsen do we plan on saving the invalid value instead?

**Comment by max-moser** [comment]
oh my, you're completely right...
that is a bit embarrassing, but i just completely ignored the response content of the `POST` call if no hard error occurred
as such, it would only ever stop at the publish action, and thus show this *apparently* inaccurate message
TL;DR: it's basically my own fault, closed!
